src/main/java/org/datadog/jenkins/plugins/datadog/util/git/GitUtils.java

```diff
@@ -259,4 +261,16 @@ public static GitClient newGitClient(final Run<?,?> run, final TaskListener list
             return null;
         }
     }
+
+    public static boolean isValidCommit(String gitCommit) {
+        if(gitCommit == null || gitCommit.isEmpty()) {
+            return false;
+        }
+
+        if(gitCommit.length() != 40) {
+            return false;
+        }
+
+        return SHA1_PATTERN.matcher(gitCommit).matches();
```

Comment:
nit but maybe this could be faster if the regex is _any non-alphanumeric_ because you're already checking length before and it could stop at the first non allowed character